src/block/Jukebox.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\tile\Jukebox as JukeboxTile;
+use pocketmine\item\Item;
+use pocketmine\item\Record;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\sound\RecordSound;
+use pocketmine\world\sound\RecordStopSound;
+
+class Jukebox extends Opaque{
+	/** @var Record|null */
+	private $record = null;
+
+	public function __construct(BlockIdentifier $idInfo, string $name, ?BlockBreakInfo $breakInfo = null){
+		parent::__construct($idInfo, $name, $breakInfo ?? new BlockBreakInfo(2.0, BlockToolType::AXE, 0, 15.0));
+	}
+
+	public function getFuelTime() : int{
+		return 300;
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($player instanceof Player){
+			if($this->hasRecord()){
+				$this->ejectRecord();
+			} else {
+				if($item instanceof Record) {
+					$player->getInventory()->remove($item);
+					$this->insertRecord($item);
+				}
+			}
+		}
+
+		$this->pos->getWorld()->setBlock($this->pos, $this);
+
+		return true;
+	}
+
+	public function hasRecord() : bool{
+		return $this->record !== null;
+	}
+
+	public function ejectRecord() : void{
+		if($this->hasRecord()){
+			$this->getPos()->getWorld()->dropItem($this->getPos()->add(0.5, 1, 0.5), $this->record);
+			$this->record = null;
+			$this->stopSound();
+		}
+	}
+
+	public function insertRecord(Record $record) : void{
+		if(!$this->hasRecord()){
+			$this->record = $record;
+			$this->startSound();
+		}
+	}
+
+	public function startSound() : void{
+		if($this->record !== null){
+			$this->getPos()->getWorld()->addSound($this->getPos(), new RecordSound($this->record->getRecordType()));
+		}
+	}
+
+	public function stopSound() : void{
+		$this->getPos()->getWorld()->addSound($this->getPos(), new RecordStopSound());
+	}
+
+	public function onBreak(Item $item, ?Player $player = null): bool
+	{
+		$this->ejectRecord();
```

Comment:
It might be better to just add this to the result of `getDrops()` so that plugins can see it in `BlockBreakEvent`.